doc/faq.mld

```diff
@@ -0,0 +1,91 @@
+important caveats. This FAQ should help you decide if it can work for you.
+
+OCamlFormat is beta software.
+While we do not follow [SemVer](https://semver.org/), we expect the program to change considerably before we reach version 1.0.0.
+In particular, upgrading the `ocamlformat` package will cause your program to get reformatted.
+Sometimes it is relatively pain-free, but sometimes it will make a diff in almost every file.
+This can be a hard price to pay, since this means losing the corresponding git history.
+
+If you use a custom configuration, options you rely on might also get removed in
+a later release.
+
+Moreover if you adopt OCamlFormat in one project it will not break your workflow in your other projects. Indeed OCamlFormat modifies a file only if it can find an `.ocamlformat` file, so adding a save hook in your editor will only simplify your workflow in projects using OCamlFormat.
+
+{2 What configuration should I use?}
+
+The recommended way is to use a versioned default profile, such as:
+
+{[
+version=0.19.0
+]}
+
+(or replace with the output of `ocamlformat --version`)
+
+This ensures two things:
+- you are using the default formatting configuration.
+- the version that you use to format is recorded somewhere.
+  If somebody else working on the project tries to use a different version,
+  they will see an error message instead of reformatting the whole project in a
+  different way.
+
+{2 Can OCamlFormat support my style?}
+
+No.
+It is better to see OCamlFormat as a tool to apply *a* style, rather than a tweakable tool to enforce your existing style.
+There are some knobs that you can turn, such as overriding `margin` to determine the maximum line width.
+But it is better not to set individual options to override what the default profile is doing.
+
+To quote (and sed) [prettier's page on option philosophy](https://prettier.io/docs/en/option-philosophy.html):
+
+> OCamlFormat has a few options because of history. **But we don’t want more of them.**
+>
+> By far the biggest reason for adopting OCamlFormat is to stop all the on-going debates over styles.
+>
+> The more options OCamlFormat has, the further from the above goal it gets. The debates over styles just turn into debates over which OCamlFormat options to use.
+
+{2 Can OCamlFormat handle Reason files?}
+
+OCamlFormat is influenced by and follows the same basic design as `refmt` for [Reason](https://github.com/facebook/reason), but outputs OCaml instead of Reason.
+
+This tool is not able to deal directly with Reason code (`*.re`/`*.rei` files),
+but it is possible to first convert these files to OCaml syntax using `refmt -p
+ml` and then running `ocamlformat` on this output.
+
+{2 How to ignore directories?}
+
+It is possible to disable OCamlFormat for the files of a directory by having an `.ocamlformat` file containing disable in this directory, or listing the files to ignore in an `.ocamlformat-ignore` file.
```

Comment:
Would be nice if it worked recursively by the way 😄 In the meantime, maybe a quick note that you need to list all the descendants of the directory to ignore them.

Reply:
done